conference.js

```diff
@@ -2336,6 +2338,102 @@ export default {
         }
     },
 
+    /**
+     * Set a device to be currently used, selected by the user.
+     *
+     * @param {MediaDeviceInfo} device - The device to save.
+     * @returns {boolean} - Returns true in order notifications to be dismissed.
+     * @private
+     */
+    _useDevice(device) {
+        switch (device.kind) {
+        case 'videoinput': {
+            APP.store.dispatch(updateSettings({
+                userSelectedCameraDeviceId: device.deviceId
+            }));
+
+            APP.store.dispatch(setVideoInputDevice(device.deviceId));
+            break;
+        }
+        case 'audioinput': {
+            APP.store.dispatch(updateSettings({
+                userSelectedMicDeviceId: device.deviceId
+            }));
+
+            APP.store.dispatch(setAudioInputDevice(device.deviceId));
+            break;
+        }
+        case 'audiooutput': {
+            setAudioOutputDeviceId(
+                device.deviceId,
+                APP.store.dispatch,
+                true)
+            .then(() => logger.log('changed audio output device'))
+            .catch(err => {
+                logger.warn(
+                    'Failed to change audio output device.',
+                    'Default or previously set audio output device will',
+                    ' be used instead.',
+                    err);
+            });
+            break;
+        }
+        }
+
+        return true;
+    },
+
+    /**
+     * Finds a new device by comparing new and old array of devices and dispatches
+     * notification with the new device.
+     * @param {MediaDeviceInfo[]} newDevices - The array of new devices we received.
+     * @param {MediaDeviceInfo[]} oldDevices - The array of the old devices we have.
+     * @private
+     */
+    _checkAndNotifyForNewDevice(newDevices, oldDevices) {
```

Comment:
I think this function can be easily moved somewhere in the react folder, isn't it?

Reply:
Same as above ^